client/app/components/ui/question.hbs

```diff
@@ -1,13 +1,9 @@
-{{#if @fieldset}}
-  <fieldset class="medium-margin-bottom" ...attributes>
-    {{yield (hash
-      Prompt=(component 'ui/fieldset-legend')
-    )}}
-  </fieldset>
-  {{else}}
-  <label class="medium-margin-bottom" ...attributes>
-    {{yield (hash
-      Prompt=(component 'ui/label-text')
-    )}}
-  </label>
-{{/if}}
+<fieldset class="ui-question" ...attributes>
+  {{yield (hash
+    questionId=this.questionId
+    Legend=(component 'ui/legend' required=@required)
+    Label=(component 'ui/label' required=@required for=this.questionId)
+    HelpText=(component 'ui/help-text')
```

Comment:
Food for thought: Here, HelpText is yielded from this component just because it is related to the Question component and other yields. From a developer standpoint, they use  `Q.HelpText` instead of `UI::HelpText` just because the help text is within a `UI::Question`. This is a nuance that may not be immediately obvious to the developer. We should add docs

Reply:
You're right. I don't think it's necessary for this to be a contextual component. Or even a component at all. I'm going to look into just making it a class.